### DISCUSSION TITLE
Override Swagger HTML?

### DISCUSSION THREAD

**Discussion by iantrudell** [strapi/strapi#7535] [Legacy Topics]
If I wanted to override the Swagger UI HTML for the Documentation plugin, e.g.

```
<title>Swagger UI</title>
<link rel="stylesheet" type="text/css" href="./swagger-ui.css" >
 <link rel="icon" type="image/png" href="./favicon-32x32.png" sizes="32x32" />
<link rel="icon" type="image/png" href="./favicon-16x16.png" sizes="16x16" />
```

or 

```
// Begin Swagger UI call region
const ui = SwaggerUIBundle({
url: "https://petstore.swagger.io/v2/swagger.json",
dom_id: '#swagger-ui',
deepLinking: true,
presets: [
    SwaggerUIBundle.presets.apis,
    SwaggerUIStandalonePreset
],
plugins: [
    SwaggerUIBundle.plugins.DownloadUrl
],
layout: "StandaloneLayout"
})
// End Swagger UI call region
```

Where would I go about doing that? I have a settings.json in /extensions/documentation/config/settings.json already but am not sure where to override static html files that are generated. Thanks!

**Comment by derrickmehaffy** [comment]
I believe this would require extending the plugin, specifically this file: https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-documentation/public/index.html

But I will ping @soupette to see if he has any thoughts

**Reply by derrickmehaffy** [reply]
After playing around it seems that file is automatically regenerated so you will have to "fork" the package and call it as a local module within your package.json:

![image](https://user-images.githubusercontent.com/8593673/90789573-ea8b9780-e2bb-11ea-8d66-92fbe79b3d39.png)

copy the `strapi-plugin-documentation` in the `plugins` directory or you can use the package.json override such as copying it into a new folder like `packages` then reference that in your package.json instead of a version

```
  "dependencies": {
    "knex": "<0.20.0",
    "sqlite3": "latest",
    "strapi": "3.1.4",
    "strapi-admin": "3.1.4",
    "strapi-connector-bookshelf": "3.1.4",
    "strapi-plugin-content-manager": "3.1.4",
    "strapi-plugin-content-type-builder": "3.1.4",
    "strapi-plugin-documentation": "file:packages/strapi-plugin-documentation",
    "strapi-plugin-email": "3.1.4",
    "strapi-plugin-upload": "3.1.4",
    "strapi-plugin-users-permissions": "3.1.4",
    "strapi-utils": "3.1.4"
  },
```

As you can see in my screenshot I just added `TESTING` string into the main body

**Reply by iantrudell** [reply]
Thanks for the reply - I'll look into this as a current solution.

Going forward though, can some customization options be added to the core plugin? It seems a bit overkill to completely override the plugin just to change the page title / logo etc.

Would be nice just to specify things like that in the core Admin UI.

**Reply by derrickmehaffy** [reply]
This plugin hasn't gotten any updates in the past few months, I believe it probably needs a bit of work to get it up to date, by all means if you want to submit a PR to make it a bit more functional for extending we would greatly appreciate it. We are a small team and are currently busy working on some other higher priority features :see_no_evil:
